src/core/DNS/SIPDNSLookupResult.cs

```diff
@@ -50,12 +55,17 @@ public class SIPDNSLookupEndPoint
         public DateTime ResolvedAt;
         public DateTime FailedAt;
         public string FailureReason;
+        public readonly DateTime ValidUntil;                // rj2: Time when DNS-Result becomes invalid, ResolveTime + TimeToLive-Seconds, Member for easy comparison
 
         public SIPDNSLookupEndPoint(SIPEndPoint sipEndPoint, uint ttl)
         {
             LookupEndPoint = sipEndPoint;
             TTL = ttl;
             ResolvedAt = DateTime.Now;
+            if (ttl != 0)
+                ValidUntil = ResolvedAt.AddSeconds(ttl);
+            else
+                ValidUntil = DateTime.MaxValue;

```

Comment:
I know I said I'm not fussy when it comes to code style, but... :) 
I can never help adding braces to if/else statements. Yes it's extra code but if it avoids a 1% chance of introducing a bug why not remove it. Heaps of literature on it https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces.

Reply:
you're right